### PR TITLE
Bring back the EventListenerOrEventListenerObject and EventListener as an extendable interface

### PR DESCRIPTION
**Pull request by orta**
[Angular](https://typescript.visualstudio.com/TypeScript/_build/results?buildId=105317&view=logs&j=71179031-6417-5a2f-3d87-af6fce2011e4&t=5ac81f15-0a74-5a9b-b155-06ef07cd84f6) relies on `EventListenerOrEventListenerObject` which makes changing this name basically a no-go I'm afraid

### CODE REVIEW

**Comment by saschanaz** [comment]
I know this is basically what the code was, but I'd prefer adding them in addedTypes.json and removing the original callback from removedTypes.json rather than complicating the build function. What do you think?

**Comment by orta** [comment]
I can give it a shot 👍🏻

**Comment by saschanaz** [comment]
I'd really like to convert those JSONs as JSON5/YAML or any other commentable formats but out of scope here 😬

**Comment by orta** [comment]
I added the ability to write comments in the JSON too, and basically apologized a bunch in the code.

Rough changes:
 - Needed to add call signatures to an interface (removed in https://github.com/saschanaz/types-web/pull/110/commits/e2f82399a6ab0a43f70762d393a0219c071013a5 and replaced with a commented hack) 
 -  Some commented why but "mother wouldn't be proud" code in the renaming process

File: inputfiles/addedTypes.json [hunk 1]
```diff
@@ -1154,6 +1189,13 @@
                     }
                 ],
                 "overrideType": "ReadableStreamDefaultReader<T>"
+            },
+            {
+                "name": "EventListenerOrEventListenerObject",
+                "type": [
+                    { "type": "_EventListener" },
```
Comment (saschanaz) [inline inputfiles/addedTypes.json:1196]: Hey, there is `override-type` to workaround this 😄

Comment (orta) [reply inputfiles/addedTypes.json]: Good point! Rebased this on main and fix that up

File: inputfiles/addedTypes.json [hunk 2]
```diff
@@ -557,6 +557,41 @@
...
+            */
+            "EventListener": {
+                "name": "EventListener",
+                "noInterfaceObject": true,
+                "methods": {
+                    "method": {
+                        // This is a hack to add a call signature, but I think it's reasonable
+                        // as it means we don't have to add a call signatures section to the
+                        // emitter for this one case.
+                        "callable": {
```
Comment (saschanaz) [inline inputfiles/addedTypes.json:573]: Clever 😄

**Review by saschanaz — CHANGES_REQUESTED** [review]

**Comment by orta** [comment]
Makes sense to me 👍🏻

File: inputfiles/knownTypes.json [hunk 3]
```diff
@@ -30,6 +31,13 @@
         "NamedCurve",
         "Pbkdf2Params",
         "PropertyIndexedKeyframes",
+        "RsaHashedImportParams",
+        "RsaHashedKeyAlgorithm",
+        "RsaHashedKeyGenParams",
+        "RsaKeyAlgorithm",
+        "RsaKeyGenParams",
+        "RsaOaepParams",
+        "RsaPssParams",
```
Comment (saschanaz) [inline inputfiles/knownTypes.json:40]: I think this one is initially sorted by `.sort()` and thus case-sensitive, but probably okay.

File: src/build/emitter.ts [hunk 4]
```diff
@@ -404,6 +405,7 @@ export function emitWebIdl(
     if (baseTypeConversionMap.has(objDomType)) {
       return baseTypeConversionMap.get(objDomType)!;
     }
+
```
Comment (saschanaz) [inline src/build/emitter.ts:408]:
```suggestion
```

**Comment by saschanaz** [comment]
Thanks! 😍

**Pull request merged**